similar/main/ai.cpp

```diff
@@ -481,11 +481,25 @@ void ai_init_boss_for_ship(void)
 #endif
 }
 
+void boss_init_all_segments(objptridx_t boss_objnum)
```

Comment:
Where possible, I prefer to have the definition use a `const` argument, e.g. `void boss_init_all_segments(const vobjptridx_t boss_objnum)` to document that the argument is never redirected to a new object.  In this context, the `const` applies only to the pointer/integer in the `objptridx_t`, but not the contained object, so it is still possible to write to the boss robot data.  If you do not need to write to it, use the `c` variant: `const vcobjptridx_t`, a constant pointer/integer that references constant data.  In this case, one of the called functions does write to the underlying `object`, so the `c` variant is not appropriate here.